src/integrated/soil_snow_model.jl

```diff
@@ -1,281 +1,435 @@
+            prognostic_land_components,
+        )
+    end
+    sources = (Soil.PhaseChange{FT}(),)
+    zero_flux = Soil.HeatFluxBC((p, t) -> 0.0)
+    boundary_conditions = (;
+        top = top_bc,
+        bottom = Soil.WaterHeatBC(;
+            water = Soil.FreeDrainage(),
+            heat = zero_flux,
+        ),
+    )
+    soil = soil_model_type{FT}(;
+        boundary_conditions = boundary_conditions,
+        sources = sources,
+        domain = domain,
+        soil_args...,
+    )
+    snow = snow_model_type(;
+        boundary_conditions = Snow.AtmosDrivenSnowBC(
+            atmos,
+            radiation,
+            prognostic_land_components,
+        ),
+        domain = Domains.obtain_surface_domain(domain),
+        snow_args...,
+    )
+
+    return LandHydrologyModel{FT, typeof(snow), typeof(soil)}(snow, soil)
+end
+
+"""
+    lsm_aux_vars(m::LandHydrologyModel)
+
+The names of the additional auxiliary variables that are
+included in the integrated Soil-Snow model.
+"""
+lsm_aux_vars(m::LandHydrologyModel) = (
+    :excess_water_flux,
+    :excess_heat_flux,
+    :atmos_energy_flux,
+    :atmos_water_flux,
+    :ground_heat_flux,
+    :effective_soil_sfc_T,
+    :sfc_scratch,
+    :subsfc_scratch,
+    :effective_soil_sfc_depth,
+)
+"""
+    lsm_aux_types(m::LandHydrologyModel)
+
+The types of the additional auxiliary variables that are
+included in the integrated Soil-Snow model.
+"""
+lsm_aux_types(m::LandHydrologyModel{FT}) where {FT} =
+    (FT, FT, FT, FT, FT, FT, FT, FT, FT)
+
+"""
+    lsm_aux_domain_names(m::LandHydrologyModel)
+
+The domain names of the additional auxiliary variables that are
+included in the integrated Soil-Snow model.
+"""
+lsm_aux_domain_names(m::LandHydrologyModel) = (
+    :surface,
+    :surface,
+    :surface,
+    :surface,
+    :surface,
+    :surface,
+    :surface,
+    :subsurface,
+    :surface,
+)
+
+"""
+    make_update_boundary_fluxes(
+        land::LandHydrologyModel{FT, SnM, SoM},
+    ) where {
+        FT,
+        SnM <: Snow.SnowModel{FT},
+        SoM <: Soil.EnergyHydrology{FT},
+        }
+
+A method which makes a function; the returned function
+updates the additional auxiliary variables for the integrated model,
+as well as updates the boundary auxiliary variables for all component
+models. 
+
+This function is called each ode function evaluation, prior to the tendency function
+evaluation.
+"""
+function make_update_boundary_fluxes(
+    land::LandHydrologyModel{FT, SnM, SoM},
+) where {FT, SnM <: Snow.SnowModel{FT}, SoM <: Soil.EnergyHydrology{FT}}
+    update_soil_bf! = make_update_boundary_fluxes(land.soil)
+    update_snow_bf! = make_update_boundary_fluxes(land.snow)
+    function update_boundary_fluxes!(p, Y, t)
+        # First compute the ground heat flux
+        # - κ_eff (T_snow - T_soil)/Δz_eff
+        # κ_eff = κ_soil * κ_snow / (κ_snow * Δz_soil / 2 + κ_soil * Δz_snow / 2)* (Δz_soil + Δz_snow)/2
+        # Δz_eff =( Δz_soil + Δz_snow)/2

```

Comment:
why is this commented out?

Reply:
It was meant as an explanation. I made a new function for the ground heat flux to clean up `update_boundary_fluxes!` and moved this commented text into the doc string for the ground heat flux function.